ttnn/cpp/ttnn/operations/ccl/ccl_common.hpp

```diff
@@ -412,17 +412,19 @@ class InterleavedRingAllGatherTensorSlicer : public LegacyCclTensorSlicer {
                     output_shape.begin() + slice_dim, output_shape.end() - 1, 1, std::multiplies<uint32_t>()) -
                 num_rows;
         } else {
-            this->num_cols = input_tensor.get_legacy_shape()[-1] / input_tensor.get_tile().get_width();
             auto input_shape = input_tensor.get_legacy_shape();
             auto output_shape = output_tensor.get_legacy_shape();
-            uint32_t num_output_cols = output_tensor.get_legacy_shape()[-1] / output_tensor.get_tile().get_width();
+            auto input_tile = input_tensor.tensor_spec().tile();
+            auto output_tile = output_tensor.tensor_spec().tile();
+            this->num_cols = input_shape[-1] / input_tensor.tensor_spec().tile().get_width();
```

Comment:
```suggestion
            this->num_cols = input_shape[-1] / input_tile.get_width();
```
I think this was missed

Reply:
Yes, thanks